src/background/services/wallet.ts

```diff
@@ -271,26 +271,26 @@ export class WalletService {
     // Note: Clear storage only if new grant type is not same as previous grant
     // type (as completeGrant already sets new grant state)
     if (existingGrants.oneTimeGrant) {
-      await this.outgoingPaymentGrantService.cancelGrant(
-        existingGrants.oneTimeGrant.continue,
-      );
       if (recurring) {
-        this.storage.set({
+        await this.storage.set({
           oneTimeGrant: null,
           oneTimeGrantSpentAmount: '0',
         });
       }
-    }
-    if (existingGrants.recurringGrant) {
       await this.outgoingPaymentGrantService.cancelGrant(
-        existingGrants.recurringGrant.continue,
+        existingGrants.oneTimeGrant.continue,
       );
+    }
+    if (existingGrants.recurringGrant) {
       if (!recurring) {
-        this.storage.set({
+        await this.storage.set({
           recurringGrant: null,
           recurringGrantSpentAmount: '0',
         });
       }
+      await this.outgoingPaymentGrantService.cancelGrant(
+        existingGrants.recurringGrant.continue,
+      );
     }
```

Comment:
This diff is difficult to reason, please check final result instead. This is what this change does:

- Add missing await on `storage.set`
- Trigger `storage.set` before `cancelGrant` (as test is looking for `storage.get`), and `cancelGrant` can take arbitrary amount of time